java/wfs-xml-query/src/main/java/com/esri/arcgisruntime/samples/wfsxmlquery/MainActivity.java

```
@@ -0,0 +1,117 @@
+        getString(R.string.wfs_feature_table_table_name));
+
+    // set the feature request mode and axis order
+    statesTable.setAxisOrder(OgcAxisOrder.NO_SWAP);
+    statesTable.setFeatureRequestMode(ServiceFeatureTable.FeatureRequestMode.MANUAL_CACHE);
+
+    statesTable.addDoneLoadingListener(() -> {
+      String xmlQuery = null;
+
+      try {
+        xmlQuery = loadQueryFromAssets();
+      } catch (IOException e) {
+        logErrorToUser(getString(R.string.error_reading_xml_file, e.getMessage()));
+      }
+
+      if (xmlQuery != null) {
+        ListenableFuture<FeatureQueryResult> featureQueryResultFuture = statesTable
+            .populateFromServiceAsync(xmlQuery, true);
+
+        // perform blocking task on a different thread to perform map operations after feature table has been populated
+        new Thread(() -> {
+          try {
+            featureQueryResultFuture.get();
+
+            // create a feature layer to visualize the table.
+            FeatureLayer statesLayer = new FeatureLayer(statesTable);
+
+            runOnUiThread(() -> {
+              // add the layer to the map.
+              mMapView.getMap().getOperationalLayers().add(statesLayer);
+
+              // set the viewpoint of the map view to the extent reported by the feature layer
+              mMapView.setViewpointGeometryAsync(statesLayer.getFullExtent(), 50);
+            });
+          } catch (InterruptedException | ExecutionException e) {
+            e.printStackTrace();
+          }
+        }).start();
+      }
+    });
+
+    statesTable.loadAsync();
+  }
+
+  private void logErrorToUser(String error) {
+    Toast.makeText(this, error, Toast.LENGTH_SHORT).show();
+    Log.e(TAG, error);
+  }
+
+  private String loadQueryFromAssets() throws IOException {
+    StringBuilder xmlString = new StringBuilder();
+    BufferedReader bufferedReader = new BufferedReader(
```

does this need a try with resources block and closed on finally?

---

I'd rather throw the exception and handle it from the calling method, but it does need closed.